lib/embed.dart

```diff
@@ -790,7 +803,57 @@ class Embed extends EditorUi {
   void displayIssues(List<AnalysisIssue> issues) {
     testResultBox.hide();
     hintBox.hide();
-    analysisResultsController.display(issues);
+
+    // Handle possiblity of issues in appended test code.
+    analysisResultsController
+        .display(detectIssuesInTestSourceAndModifyIssuesAccordingly(issues));
+  }
+
+  // We append test source code to the user's source code, because of
+  // this we possibly have a special situation..
+  // There could be warnings or errors in the *TEST* code that is being
+  // appended to the user's dart source.
+  // This can result in issues with line numbers that are
+  // outside the user's dart source.  This would confusing to the users.
+  // We are going to do one of two things:
+  // - If the test source is currently HIDDEN and the issue kind is
+  // not and `error` (it is `info` or `warning`) then we will REMOVE
+  // the issue from the list so as to "hide" it.
```

Comment:
I'm not sure we should hide analysis warnings in the test code, even if the test code is hidden. By default we hide it, and it would make it difficult to figure out if a codelab needs to be fixed (you'd have to manually go through the codelab and click "show test code")

Reply:
My reasoning here was that I encountered several of the flutter exercises at the I/O that had a list of issues within the test code, and that test code of course had line numbers past the end of the source...  These examples were presumably for new or less experienced dart/flutter users and it seemed like it would be very confusing to the user to see a list of issues that were not in code that they were even aware of.
   I thought it would be better in this case to hide warnings in the test code as they would only confuse the user and they definitely took away from a 'polished' feel to the catch-a-dart exercises at I/O.   The author having to periodically review (when there were analysis/linter rule changes) the test code for new warnings by 'show test code' seems less intrusive that all users having to see the confusing warnings in code past the end of what they can see, in test code they are probably unaware of.. 
   Anyway, that was my thinking here... lol
   I do think it is a conversation worth having.
   I also pondered some other solutions, like somehow being able to notify someone/something when exercises/examples 'age' in this manner and start having issues like this.  It seems like there must be some solution that is better than confusing the user ?

Reply:
cc: @domesticmouse, I believe our tests make sure that codelabs don't have analysis warnings, so we might want to clean those up if we aren't catching those up-front. 

As a codelab author, I would probably prefer to see the warnings so I can fix the broken test code, but as someone following the codelab this is definitely not great.

Reply:
We have zero testing on the gist based code. As for workshops, I see new linter warnings every quarter, and our community contributed workshops don't always have CI.

Reply:
(bear with me - thought exercise here)
Perhaps we could make it easier for the codelab/exercise writer to 'show test code' for all dartpads ?

Something like having inject_embed.dart check for a cookie or a flag in the URL - and then automatically add a flag to the url of all embedded dartpads to '&showtestcode=true'

A flag would be simple to add to the parent doc url for the writer, but they would need to do that each time.
A cookie flag would be nice because it could be set on a global basis for https://dart.dev/ (for example), but we would be a way to set it (perhaps by setting a url flag and we would set the cookie when we detect that).  Once the writer set it then it would persist.

Reply:
Can we utilise the new multi file capability in the backend to improve this situation.